### PR TITLE
feat(sucrase): resolve .tsx files

### PR DESCRIPTION
**Pull request by flipsasser**
## Rollup Plugin Name: `sucrase`

This PR contains:

- [x] bugfix
- [ ] feature
- [ ] refactor
- [ ] documentation
- [ ] other

Are tests included?

- [x] yes (_bugfixes and features will not be merged without tests_)
- [ ] no

Breaking Changes?

- [ ] yes (_breaking changes will not be merged unless absolutely necessary_)
- [x] no

### Description

The `@rollup/plugin-sucrase` plugin does not resolve `.tsx` files. This PR patches that behavior to allow the use of `@rollup/plugin-sucrase` in React / TypeScript projects.

It includes a test that imports a TSX file and snapshots the JSX output. It works. Which is great!

Here's a different way of saying it:

- Expected behavior: using `@rollup/plugin-sucrase` with `transforms: ["typescript", "jsx"]` will correctly import both `.ts` and `.tsx` files.
- Actual behavior: `@rollup/plugin-sucrase` cannot resolve `.tsx` files; only `.ts`.

### CODE REVIEW

File: packages/sucrase/src/index.js [hunk 1]
```diff
@@ -15,9 +15,11 @@ module.exports = function sucrase(opts = {}) {
       if (importer && /^[./]/.test(importee)) {
         const resolved = path.resolve(importer ? path.dirname(importer) : process.cwd(), importee);
         // resolve in the same order that TypeScript resolves modules
-        const resolvedFilename = [`${resolved}.ts`, `${resolved}/index.ts`].find((filename) =>
-          fs.existsSync(filename)
-        );
+        const resolvedFilename = [
```
[the remark this answers is not in the document]
Comment (shellscape) [reply packages/sucrase/src/index.js:18]: @flipsasser please take a look at this comment

[the remark this answers is not in the document]
Comment (flipsasser) [reply packages/sucrase/src/index.js:18]: Hey sorry somehow I missed this - will fix. Thanks for the review!

[the remark this answers is not in the document]
Comment (flipsasser) [reply packages/sucrase/src/index.js:18]: It now follows the same pattern as ts imports

**Comment by flipsasser** [comment]
@Andarist what is the NPM release schedule like, and can I assist in getting this update published? Please let me know if I can help. Thanks!

**Pull request merged**